pkg/nfd-master/nfd-master.go

```diff
@@ -228,8 +244,18 @@ func filterFeatureLabels(labels Labels, extraLabelNs []string, labelWhiteList *r
 			stderrLogger.Printf("%s does not match the whitelist (%s) and will not be published.", name, labelWhiteList.String())
 			delete(labels, label)
 		}
+
+		// Skip labels which are intended to be extended resources
+		if _, ok := extendedResources[label]; ok {
+			if _, err := strconv.Atoi(labels[label]); err != nil {
+				stderrLogger.Printf("bad label value encountered for extended resource: %s", err.Error())
+				continue // non-numeric label can't be used
+			}
+			extendedResources[label] = labels[label] // store value
+			delete(labels, label)
+		}
```

Comment:
I think you could do this filtering the other way around. In a separate loop, go through `extendedResourceNames`, delete labels that match and add them to `extendedResources` map.

This would also eliminate the need to initialize `extendedResources` above. And, as `extendedResources` would only have legit values, you could possibly omit some `value != ""` checks in other functions.